tools/autograd/templates/VariableType.cpp

```diff
@@ -122,7 +122,8 @@ Tensor & VariableType::unpack_any(const Tensor & t, const char * name, int pos)
         pos, name);
   }
   auto scalarType = t.type().scalarType();
-  auto& type = *VariableImpl::getType(baseType->toScalarType(scalarType));
+  auto backend = t.type().backend();
+  auto& type = *VariableImpl::getType(baseType->toScalarType(scalarType).toBackend(backend));
```

Comment:
If this `toBackend` could incur CUDA copies, you need to ensure you're on the correct device (or that all call sites set the device before hand - please add a comment in this case)

Reply:
this is a bit confusing, but there is toBackend on a Tensor that can create CUDA copies and toBackend on a type that gives you the type corresponding to a possibly different backend (just looks it up in a table).  This is operating on the Type.

Reply:
Oh, right. Didn't notice it's only looking up the type and using this as a no-op check. Never mind the comment